src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksTaxRuleDependencyServiceImpl.java

```diff
@@ -0,0 +1,261 @@
+package edu.cornell.kfs.pmw.batch.service.impl;
+
+import java.sql.Date;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.vnd.VendorConstants;
+import org.kuali.kfs.vnd.businessobject.VendorHeader;
+import org.kuali.rice.core.api.config.property.ConfigurationService;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+
+import edu.cornell.kfs.pmw.batch.PaymentWorksConstants;
+import edu.cornell.kfs.pmw.batch.PaymentWorksKeyConstants;
+import edu.cornell.kfs.pmw.batch.TaxRule;
+import edu.cornell.kfs.pmw.batch.businessobject.KfsVendorDataWrapper;
+import edu.cornell.kfs.pmw.batch.businessobject.PaymentWorksIsoFipsCountryItem;
+import edu.cornell.kfs.pmw.batch.businessobject.PaymentWorksVendor;
+import edu.cornell.kfs.pmw.batch.service.PaymentWorksBatchUtilityService;
+import edu.cornell.kfs.pmw.batch.service.PaymentWorksTaxRuleDependencyService;
+import edu.cornell.kfs.sys.CUKFSConstants;
+
+public class PaymentWorksTaxRuleDependencyServiceImpl implements PaymentWorksTaxRuleDependencyService {
+    private static final Logger LOG = LogManager.getLogger(PaymentWorksTaxRuleDependencyServiceImpl.class);
+    
+    protected DateTimeService dateTimeService;
+    protected PaymentWorksBatchUtilityService paymentWorksBatchUtilityService;
+    protected ConfigurationService configurationService;
+    private DateTimeFormatter dateTimeFormatter;
+    
+
+    @Override
+    public KfsVendorDataWrapper populateTaxRuleDependentAttributes(PaymentWorksVendor pmwVendor,
+            Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap) {
+        LOG.info("populateTaxRuleDependentAttributes, entering");
+        KfsVendorDataWrapper vendorDataWrapper = new KfsVendorDataWrapper();
+        addVendorHeaderToKfsVendorDataWrapper(pmwVendor, paymentWorksIsoToFipsCountryMap, vendorDataWrapper);
+        return vendorDataWrapper;
+    }
+
+    protected void addVendorHeaderToKfsVendorDataWrapper(PaymentWorksVendor pmwVendor,
+            Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap, KfsVendorDataWrapper vendorDataWrapper) {
+        String vendorFipsCountryCode = convertIsoCountryCodeToFipsCountryCode(pmwVendor.getRequestingCompanyTaxCountry(), paymentWorksIsoToFipsCountryMap);
+        TaxRule taxRule = determineTaxRuleToUseForDataPopulation(pmwVendor, vendorFipsCountryCode);
+        
+        vendorDataWrapper.getVendorDetail().getVendorHeader().setVendorForeignIndicator(!isUnitedStatesFipsCountryCode(vendorFipsCountryCode));
```

Comment:
would it be better to get the Vendor header once and then populate the needed attributes instead of calling the getters so many times?

Reply:
sure

Reply:
hmmm, I wonder how I lost that, put back in though